### ISSUE TITLE
Generic.PHP.UpperCaseConstant: don't check null and false types

### ISSUE THREAD

**Issue by KacerCZ** [squizlabs/PHP_CodeSniffer#3332]
**Describe the bug**
Sniff `Generic.PHP.UpperCaseConstant` checks all occurences of FALSE/NULL even if they are used in type.

**Code sample**
```php
function debug(string|null $message): void|false {
        echo "DEBUG: $message\n";
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <rule ref="Generic.PHP.UpperCaseConstant" />
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
-------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 1 LINE
--------------------------------------------------------------------------------------------
 3 | ERROR | [x] TRUE, FALSE and NULL must be uppercase; expected "NULL" but found "null"
 3 | ERROR | [x] TRUE, FALSE and NULL must be uppercase; expected "FALSE" but found "false"
```

**Expected behavior**
No error is reported for null/false used as type.

**Versions (please complete the following information):**
 - OS: Ubuntu Linux 20.10
 - PHP: 8.0.5
 - PHPCS: 3.6.0
 - Standard: custom

**Comment by KacerCZ** [comment]
Sniff `Generic.PHP.UpperCaseConstant` also works in opposite to sniff `Generic.PHP.LowerCaseType`, which correctly recognizes when `false` and `null` are used in type definition and not as values.

**Comment by jrfnl** [comment]
@KacerCZ Thanks for reporting this. PR #3429 should fix this. Testing appreciated.

**Comment by KacerCZ** [comment]
@jrfnl I tested your patch and it fixes reported issue. Thank you.

**Comment by jrfnl** [comment]
Thanks for testing!

**Comment by gsherwood** [comment]
Moved to 3.7.0 as this is a BC break.

I think this change can be justified even though the purpose of the sniff is clear. The justification is that return types didn't exist when the sniff was created, so the intention was originally to just enforce the use of true/false/null as values. So I think the change is a good one, but it could still leave some developers without an uppercase enforcement of type values.
